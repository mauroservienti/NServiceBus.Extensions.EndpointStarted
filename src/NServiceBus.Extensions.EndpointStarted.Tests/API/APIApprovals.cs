using System.Runtime.CompilerServices;
using NUnit.Framework;
using PublicApiGenerator;
using VerifyNUnit;

namespace NServiceBus.Extensions.EndpointStarted.Tests.API
{
    public class APIApprovals
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve_API()
        {
            var publicApi = typeof(OnEndpointStartedEndpointConfigurationExtensions).Assembly.GeneratePublicApi(new ApiGeneratorOptions
            {
                ExcludeAttributes = new[]
                {
                    "System.Runtime.Versioning.TargetFrameworkAttribute"
                }
            });
            Verifier.Verify(publicApi);
        }
    }
}