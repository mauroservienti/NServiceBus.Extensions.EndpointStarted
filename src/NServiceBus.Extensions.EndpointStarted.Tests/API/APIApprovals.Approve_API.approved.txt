[assembly: System.Reflection.AssemblyMetadata("RepositoryUrl", "https://github.com/mauroservienti/NServiceBus.Extensions.EndpointStarted.git")]
[assembly: System.Runtime.Versioning.TargetFramework(".NETCoreApp,Version=v6.0", FrameworkDisplayName=".NET 6.0")]
namespace NServiceBus
{
    public static class OnEndpointStartedEndpointConfigurationExtensions
    {
        public static void OnEndpointStarted(this NServiceBus.EndpointConfiguration configuration, System.Func<NServiceBus.IMessageSession, System.Threading.Tasks.Task> onEndpointStarted) { }
        public static void OnEndpointStarted(this NServiceBus.EndpointConfiguration configuration, System.Func<NServiceBus.IMessageSession, System.IServiceProvider, System.Threading.Tasks.Task> onEndpointStarted) { }
    }
}